oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -102,6 +118,10 @@ private UserAuthorizer(
     this.userAuthUri = (userAuthUri == null) ? OAuth2Utils.USER_AUTH_URI : userAuthUri;
     this.tokenStore = (tokenStore == null) ? new MemoryTokensStorage() : tokenStore;
     this.pkce = pkce;
+    this.clientAuthenticationType =
+        (clientAuthentication == null)
+            ? ClientAuthenticationType.CLIENT_SECRET_POST
+            : clientAuthentication;
```

Comment:
Just to confirm my understanding, the default ClientAuthenticationType would be `CLIENT_SECRET_POST`, right? This would be the type if a user doesn't explicitly specify it/ set it via the setter?

And `ClientAuthenticationType.NONE` has a different behavior and needs to be explicitly set if they want to use NONE.

Reply:
Correct. The `CLIENT_SECRET_POST` is what the current behavior following, so that should be the default.

And `NONE` shall be explicitly set. It throw errors naturally since on backend we need some sort of basic auth either having `client_secret` in post or having `authorization` in the headers.